modules/apps/redirect/redirect-web/src/main/resources/META-INF/resources/js/RedirectManagementToolbarPropsTransformer.js

```diff
@@ -12,28 +12,33 @@
  * details.
  */
 
-import {DefaultEventHandler} from 'frontend-js-web';
-import {Config} from 'metal-state';
+import {postForm} from 'frontend-js-web';
 
-class RedirectManagementToolbarDefaultEventHandler extends DefaultEventHandler {
-	deleteSelectedRedirectEntries() {
-		const form = this.one('#fm');
+export default function propsTransformer({
+	additionalProps: {deleteRedirectEntriesURL},
+	portletNamespace,
+	...otherProps
+}) {
+	const deleteSelectedRedirectEntries = () => {
+		const form = document.getElementById(`${portletNamespace}fm`);
 
-		Liferay.Util.postForm(form, {
+		postForm(form, {
 			data: {
 				deleteEntryIds: Liferay.Util.listCheckedExcept(
 					form,
-					this.ns('allRowIds')
+					`${portletNamespace}allRowIds`
 				),
 			},
-			url: this.deleteRedirectEntriesURL,
+			url: deleteRedirectEntriesURL,
 		});
-	}
-}
-
-RedirectManagementToolbarDefaultEventHandler.STATE = {
-	deleteRedirectEntriesURL: Config.string(),
-	spritemap: Config.string(),
-};
+	};
 
-export default RedirectManagementToolbarDefaultEventHandler;
+	return {
+		...otherProps,
+		onActionButtonClick: (event, {item}) => {
+			if (item?.data?.action === 'deleteSelectedRedirectEntries') {
```

Comment:
Looking at [this](https://github.com/liferay-frontend/liferay-portal/blob/5143e1ca77ded386e0edb3d1f9c1069c09b51290/modules/apps/redirect/redirect-web/src/main/java/com/liferay/redirect/web/internal/display/context/RedirectManagementToolbarDisplayContext.java#L66-L75) makes me that think `item`,  `item.data` and `item.data.action` are always going to be defined, so I don't think there's a need to use optional chaining (`?.`) ... am I right?

Reply:
You are right, it should always be defined, I'll change it.